src/app/app.page/app.page.component.html

```diff
@@ -1831,9 +1850,42 @@ <h4 class="card-title text-left" i18n="@@newLiveStreamCardTitle">
+                                                                <input [(ngModel)]="appSettings.jwtControlEnabled"
+                                                                   id="jwtFilter"
+                                                                   name="jwtFilter" type="checkbox">
+
+                                                                <label for="jwtFilter"
+                                                                   i18n="@@jwtFilter">
+                                                                Enable JWT Filter<a
+                                                                    matTooltip="If you disable JWT Filter, we are strongly recommending enable IP Filter."><i
+                                                                    class="fa fa-info-circle"></i></a>
+                                                                </label>
+                                                            </div>
+                                                        </div>
 
-                                                    </div>
+                                                        <div class="col-xs-12" style="margin-top:12px">
+                                                            <label i18n="@@jwtFilter">
+                                                                JWT Filter Secret Key<a
+                                                                    matTooltip="Please enter at least 32 character"><i
+                                                                    class="fa fa-info-circle"></i></a>
+                                                            </label>
+                                                            <div>
+                                                                <textarea name="jwtSecret" minlength="32"  #jwtSecret="ngModel" [disabled]="!appSettings.jwtControlEnabled"
+                                                                          [(ngModel)]="appSettings.jwtSecretKey"
+                                                                          type="textarea" class="form-control"
+                                                                          placeholder="Please write at least 32 character"
+                                                                          rows="1">{{appSettings.jwtSecretKey}}
+                                                                </textarea>
+                                                                <small [hidden]="jwtSecret.valid || !appSettings.jwtControlEnabled"
+                                                                       class="text-danger">
+                                                                    JWT Secret Key should at least 32 character
+                                                                </small>
+                                                            </div>
+                                                        </div>
```

Comment:
If JWT filter is enabled but no secret key is provided, click the save button, it does not save the settings and does not show any warning